channels_redis/core.py

```diff
@@ -628,25 +642,30 @@ async def group_send(self, group, message):
         ) = self._map_channel_keys_to_connection(channel_names, message)
 
         for connection_index, channel_redis_keys in connection_to_channel_keys.items():
+            # Discard old messages based on expiry
+            for key in channel_redis_keys:
+                await connection.zremrangebyscore(
+                    key, min=0, max=int(time.time()) - int(self.expiry)
+                )
 
             # Create a LUA script specific for this connection.
             # Make sure to use the message specific to this channel, it is
             # stored in channel_to_message dict and contains the
             # __asgi_channel__ key.
 
-            group_send_lua = (
-                """ local over_capacity = 0
+            group_send_lua = """ local over_capacity = 0
                     for i=1,#KEYS do
-                        if redis.call('LLEN', KEYS[i]) < tonumber(ARGV[i + #KEYS]) then
-                            redis.call('LPUSH', KEYS[i], ARGV[i])
+                        if redis.call('ZCOUNT', KEYS[i], '-inf', '+inf') < tonumber(ARGV[i + #KEYS]) then
+                            redis.call('ZADD', KEYS[i], %f, ARGV[i])
                             redis.call('EXPIRE', KEYS[i], %d)
                         else
                             over_capacity = over_capacity + 1
                         end
                     end
                     return over_capacity
-                    """
-                % self.expiry
+                    """ % (
+                time.time(),
+                self.expiry,
```

Comment:
Did this need adjusting per the comment on the issue? (Should we handle it separately?)

Reply:
@carltongibson good point, will fix that later today

Reply:
@astutejoe Super thank. Throw it on top here as a separate commit (it's probably worth calling out separately.) If you fancy adding a line (or two) to CHANGELOG.txt that's cool too. (I can do it if not.) 

Thanks again!

Reply:
@carltongibson There you go my dude, hope it's up to the standards. Feel free to change the CHANGELOG if the language isn't suitable or precise.